src/ServicePulse.Host/vue/src/components/failedmessages/OrderBy.vue

```diff
@@ -31,14 +39,18 @@ function loadSavedSortOption() {
     const sortBy = getSortOptions().find((sort) => {
       return sort.description.toLowerCase() === criteria.toLowerCase();
     });
-    return { sort: getSortFunction(sortBy.selector, direction), dir: direction, description: sortBy.description };
+    if (sortBy) {
+      sortBy.sort = getSortFunction(sortBy.selector, direction);
+      sortBy.dir = direction;
+      return sortBy;
+    }
   }
 
   return props.sortOptions[0];
 }
 
-function getSortFunction(selector, dir) {
-  return (firstElement, secondElement) => {
+function getSortFunction(selector: SortOptions["selector"], dir: string) {
+  return (firstElement: GroupType, secondElement: GroupType) => {
     if (dir === "asc") {
```

Comment:
the direction strings can be extracted to an enum

Reply:
Done